src/services/api/routes/communities.ts

```diff
@@ -28,22 +28,48 @@ const createNewCommunitySchema = z.object({
 
 const router: express.Router = express.Router();
 
+async function respondCommunityNotFound(response: express.Response) : Promise<void> {
+	response.status(404);
+	response.send(xmlbuilder.create({
+		result: {
+			has_error: 1,
+			version: 1,
+			code: 404,
+			error_code: 919,
+			message: 'COMMUNITY_NOT_FOUND'
+		}
+	}).end({ pretty: true }));
+}
+
+async function commonGetSubCommunity(request: express.Request, response: express.Response, communityID: string | undefined) : Promise<HydratedCommunityDocument | null> {
+
+	const parentCommunity: HydratedCommunityDocument | null = await getCommunityByTitleID(request.paramPack.title_id);
+	if (!parentCommunity) {
+		await respondCommunityNotFound(response);
+		return parentCommunity;
```

Comment:
Only going to comment this once because it appears in multiple places, but returning `parentCommunity` here feels a little odd? I know that in these branches `parentCommunity` is null, so you're effectively just doing `return null`, but when glancing over the code it makes it look like `parentCommunity` has a value since it's being returned instead of an explicit `null` return